src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentMarkupEntityPass.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Text;
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
+
+namespace Microsoft.AspNetCore.Razor.Language.Components
+{
+    internal class ComponentMarkupEntityPass : ComponentIntermediateNodePassBase, IRazorOptimizationPass
```

Comment:
`ComponentMarkupRewritePass`? I don't think this name quite captures all of it.

Reply:
`ComponentMarkupRewritePass` seems too generic, do you like `ComponentMarkupEncodingPass` better?

Reply:
sounds good